pkg/bff/db/models/v1/announcements.go

```diff
@@ -0,0 +1,53 @@
+package models
+
+import (
+	"sort"
+	"time"
+)
+
+const (
+	PostDateLayout = "2006-01-02"
+	MonthLayout    = "2006-01"
+)
+
+// Month returns the postdate month in the form YYYY-MM to determine which
+// AnnouncementsMonth the announcement should belong in.
+func (a *Announcement) Month() (_ string, err error) {
+	var postDate time.Time
+	if postDate, err = a.ParsePostDate(); err != nil {
+		return "", err
+	}
+	return postDate.Format(MonthLayout), nil
+}
+
+// Return the timestamp from the post date.
+func (a *Announcement) ParsePostDate() (time.Time, error) {
+	return time.Parse(PostDateLayout, a.PostDate)
+}
+
+// Add an announcement ensuring that they are stored sorted by post date.
+// NOTE: can sort postdate strings in the YYYY-MM-DD format without parsing them,
+// however the post date must be validated before adding it to the month.
```

Comment:
If there are two posts on the same day, they're inserted into the `AnnouncementMonth` in sorted order; I believe that Go implements left-side binary search but later posts will either always be inserted to the left side of the previous post(s) or to the right side of the previous post(s). So the ordering should be consistent because we're using insort:

https://go.dev/play/p/_3O9w1CcDnO

It looks like if we make a Post at 10:00 am and then a post at 3:00 pm -- the 10am post will be first in the queue; I guess it could become an issue but it's probably fine for now (famous last words).

Reply:
I did think about using the `modified` timestamp, but if we use that then the post date may not match the order, e.g. if we update the announcement the next day.